packages/web3-eth-contract/types/index.d.ts

```diff
@@ -0,0 +1,191 @@
+    ): void;
+    once(
+        event: string,
+        options: EventOptions,
+        callback: (error: Error, event: EventData) => void
+    ): void;
+
+    events: any;
+
+    getPastEvents(event: string): Promise<EventData[]>;
+    getPastEvents(
+        event: string,
+        options: EventOptions,
+        callback: (error: Error, event: EventData) => void
+    ): Promise<EventData[]>;
+    getPastEvents(event: string, options: EventOptions): Promise<EventData[]>;
+    getPastEvents(
+        event: string,
+        callback: (error: Error, event: EventData) => void
+    ): Promise<EventData[]>;
+}
+
+export class ContractModuleFactory {} // TODO: Define methods
+
+export interface Options {
+    address: string;
+    data: string;
+}
+
+export interface DeployOptions {
+    data: string;
+    arguments?: any[];
+}
+
+export interface ContractSendMethod {
+    send(
+        options: SendOptions,
+        callback?: (err: Error, transactionHash: string) => void
+    ): PromiEvent<Contract>;
+
+    estimateGas(
+        options: EstimateGasOptions,
+        callback?: (err: Error, gas: number) => void
+    ): Promise<number>;
+
+    estimateGas(callback: (err: Error, gas: number) => void): Promise<number>;
+
+    estimateGas(
+        options: EstimateGasOptions,
+        callback: (err: Error, gas: number) => void
+    ): Promise<number>;
+
+    estimateGas(options: EstimateGasOptions): Promise<number>;
+
+    estimateGas(): Promise<number>;
+
+    encodeABI(): string;
+}
+
+export interface SendOptions {
+    from: string;
+    gasPrice?: string;
+    gas?: number;
+    value?: number | string | BN;
+}
+
+export interface EstimateGasOptions {
+    from?: string;
+    gas?: number;
+    value?: number | string | BN;
+}
+
+export interface ContractOptions extends Web3ModuleOptions {
+    // Sender to use for contract calls
+    from?: string;
+    // Gas price to use for contract calls
+    gasPrice?: string;
+    // Gas to use for contract calls
+    gas?: number;
+    // Contract code
+    data?: string;
+}
+
+export interface EventOptions {
+    filter?: any;
+    fromBlock?: number;
+    toBlock?: string | number;
+    topics?: any[];
+}
+
+export interface EventData {
+    returnValues: {
+        [key: string]: any;
+    };
+    raw: {
+        data: string;
+        topics: string[];
+    };
+    event: string;
+    signature: string;
+    logIndex: number;
+    transactionIndex: number;
+    transactionHash: string;
+    blockHash: string;
+    blockNumber: number;
+    address: string;
+}
+
+export interface AbiModel {
+    getMethod(name: string): AbiItemModel | false;
+
+    getMethods(): AbiItemModel[];
+
+    hasMethod(name: string): boolean;
+
+    getEvent(name: string): AbiItemModel | false;
+
+    getEvents(): AbiItemModel[];
+
+    getEventBySignature(signature: string): AbiItemModel;
+
+    hasEvent(name: string): boolean;
+}
+
+export interface AbiItemModel {
```

Comment:
The class AbiItemModel doesn't exist in 1.0

Reply:
it does exist look at web3-eth-contract line 107 it settings that and mapping values. I think i just need to remove the functions on it and remove `anonymous`

Reply:
```ts
export interface AbiItemModel {
    signature: string;
    name: string;
    payable: boolean;
}
```